src/overlap_check/test/overlap_check_test.cpp

```diff
@@ -0,0 +1,113 @@
+#include <iostream>
+#include <cmath>
+#include <cstdlib>
+#include <ctime>
+#include <vector>
+#include <set>
+#include <algorithm>
+#include <memory>
+#include "moab/Core.hpp"
+
+#include "overlap.hpp"
+
+#include "overlap_check_test.hpp"
+
+void OverlapTest::SetUp() {
+  SetFilename();
+
+  MBI = std::shared_ptr<Interface>(new Core());
+
+  ErrorCode rval = MBI->load_file(filename.c_str());
+  EXPECT_EQ(rval, MB_SUCCESS);
+};
+
+void OverlapTest::TearDown() {
+  ErrorCode rval = MBI->delete_mesh();
+  EXPECT_EQ(rval, MB_SUCCESS);
+};
+
+class OverlappingVolumesTest : public OverlapTest {
+  virtual void SetFilename() { filename = "overlap.h5m"; }
```

Comment:
This and all other SetFilename functions below should be marked with `override`. eg.:
```c++
    virtual void SetFilename() override { filename = "overlap.h5m"; }
```
This is more of a discipline that is good to follow as it makes it more likely that mistakes with virtual/overridden functions will result in compiler errors, making them easier to address.